entity-type-service-rx-client/src/main/java/org/hypertrace/entity/type/service/rxclient/EntityTypeCachingClient.java

```diff
@@ -48,10 +44,9 @@ public Observable<EntityType> getAll() {
   }
 
   @Override
-  public Single<EntityType> get(String name) {
+  public Single<Optional<EntityType>> get(String name) {
```

Comment:
Unnecessary breaking change. Don't change the API of a published lib. Caller can always catch so this doesn't unlock any new functionality anyway. I'm guessing you're trying to differentiate between an exception due to a missing type vs a downstream error? If so, just change the exception thrown to a not_found. Upstream, handle not found differently than other errors